src/dataframe/dataframe.jl

```diff
@@ -116,8 +117,13 @@ mutable struct DataFrame <: AbstractDataFrame
                 throw(DimensionMismatch("columns must be 1-dimensional"))
             end
         end
-        new(columns, colindex)
+        new(columns, colindex, MetaData())
     end
+
+    function DataFrame(columns::Vector{Any}, colindex::Index, metadata::MetaData)
```

Comment:
Replace this with `metadata::MetaData=MetaData()` in the constructor above. Here you're bypassing all consistency checks done by the existing constructor.